### ISSUE TITLE
Make a haxelib release

### ISSUE THREAD

**Issue by maxless** [tong/hxssl#38]
Haxelib still has 3.0.0-alpha of 2013.

**Comment by Blank101** [comment]
Agreed. Major improvements have been made since then. I would if I could, but I do not have access to do haxelib release. Ping @tong

**Comment by stahlratte** [comment]
yes, please update the release. :) @tong

**Comment by tong** [comment]
@Blank101 please tell me your haxelib username so i can give you access.
(Sorry for being that lame on updates, this project is atm light years away from what i am focusing right  now.)

**Comment by Blank101** [comment]
Blank101 is my haxelib user.

**Comment by tong** [comment]
Adding you as haxelib contributor (in haxelib.json) doesn't give you write access to haxelib, does it?
Anyhow i can upload a new version when all ndll files got updated. Currently only linux64.ndll is uptodate and i cannot provide builds for other platforms.

**Comment by Blank101** [comment]
I'm not sure about the haxelib contributor thing. I'll see what I can do about the other builds. Who was normally building the other ndlls before?

**Comment by delahee** [comment]
I used to do it but I would gladly pass the rights  :)
What do you require me to do ?

**Comment by Blank101** [comment]
In order to release we need Mac 64, Linux 32 and Windows (32 bit I assume?) builds done. Raspberry PI can be done too as a bonus, but I don't really think it's a necessity.

**Comment by delahee** [comment]
I think I can provide the win32 and mac 64 build for once. Lin32 and RPI we
don't have the hardware here. The library is fairly easy to rebuild so I
think letting them hang is no problem.

I ll try to do it next week.



2015-04-23 18:29 GMT+02:00 Sam <notifications@github.com>:

> In order to release we need Mac 64, Linux 32 and Windows (32 bit I
> assume?) builds done. Raspberry PI can be done too as a bonus, but I don't
> really think it's a necessity.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tong/hxssl/issues/38#issuecomment-95644659>.
>



-- 
David Elahee

**Comment by tong** [comment]
Updated RPi ndll.

**Comment by Blank101** [comment]
Mac 64 has been updated. All we need is Win32, and we can release.

**Comment by yupswing** [comment]
i can do win32 tomorrow morning if you need it

**Comment by Blank101** [comment]
Looks like this is good to go for a release now.

**Comment by hughsando** [comment]
Did the new code run ok?  I got some certificate errors when testing.

**Comment by Blank101** [comment]
Which platform are you getting certificate errors on? Seems to work fine on Linux.

**Comment by hughsando** [comment]
Windows and mac, with the new binaries.  If I used the 3.0.0 alpha, it
seemed to work.

**Comment by Blank101** [comment]
What url are you hitting?

**Comment by hughsando** [comment]
Just did a fresh checkout on  a windows box.  It appears I need the line ```{name:"objects.h", src:"crypto/objects/objects.h"},``` in Build.hx not to be commented out to compile.

Running sample 01:
```
>./test-securesocket.exe
Connecting to : talk.google.com ...
10776:error:14090086:SSL routines:ssl3_get_server_certificate:certificate verify failed:./ssl/s3_clnt.c:1205:
Error : ERROR
```

Running sample 02: 
```
>./test.exe
Error : Failure type not string @ ./hxssl_ssl.cpp:307
```

 Running sample 03:  
```
>./test.exe
7400:error:14077438:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert internal error:./ssl/s23_clnt.c:769:
ERROR
```

And test:
```
>./Run.exe
Class: TestBase64 ...
Class: TestMD5 .
Class: TestSHA1 .
Class: TestSHA256 .
Class: TestRIPEMD160 .
Class: TestSocket 13168:error:14090086:SSL routines:ssl3_get_server_certificate: certificate verify failed:./ssl/s3_clnt.c:1205:
E
* TestSocket::test_connect_ssl()
ERR: exception thrown : ERROR
```

**Comment by Blank101** [comment]
Will look into this. I haven't tested the examples recently. Those should probably be updated since there have been a lot of changes. The certificate verify failure on the unit tests is definitely a concern though.

3.0.0 alpha works because it does not do certificate validation which is fairly insecure.

**Comment by hughsando** [comment]
Yes, but no certification has its uses too.  Maybe need to add some mode to skip or supply a certificate.

**Comment by Blank101** [comment]
Yes. You can set validateCert to false on the tls socket to skip this.

**Comment by Blank101** [comment]
@hughsando When you get a chance can you re-test this on Windows with latest Haxe and hxssl builds? I believe I've managed to fix the problem.

**Comment by Justinfront** [comment]
What's the mac status?  I tried the 2nd example on mac 10.11.3 with git version, using current haxe.
Struggled to get c++ to work and neko when I run the client
```code
MacBook-Pro:TestiHssl2 justinfront$ ./test
Error : Failure type not string @ ./hxssl_ssl.cpp:300
MacBook-Pro:TestiHssl2 justinfront$ neko test.n
TestSecureSocketServer.hx:16: Starting server...
Called from ? line 1
Called from TestSecureSocketServer.hx line 18
Called from sys/ssl/Socket.hx line 316
Uncaught exception - (1) : An unknown error has occurred.
MacBook-Pro:TestiHssl2 justinfront$ neko test.n
TestSecureSocketServer.hx:16: Starting server...
Called from ? line 1
Called from TestSecureSocketServer.hx line 18
Called from sys/ssl/Socket.hx line 316
Uncaught exception - (1) : An unknown error has occurred.
```
and 
```code
Called from ? line 1
Called from Client.hx line 14
Called from sys/ssl/Socket.hx line 194
Called from sys/ssl/Socket.hx line 185
Uncaught exception - (1) : An unknown error has occurred.
```
Is there something I am missing.  I ran 
```neko test.n```
in one terminal and in the other 
```neko client.n```
